src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,444 @@
+
+    /**
+     * Prints a border made of "_" characters
+     */
+    private static void printBorder() {
+        System.out.println("____________________________________________________________");
+    }
+
+    private static void printGreetingMessage() {
+        printBorder();
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        printBorder();
+    }
+
+    private static void printGoodbyeMessage() {
+        printBorder();
+        System.out.println("Bye. Hope to see you again soon!");
+        printBorder();
+    }
+
+    /**
+     * Prints the error message for an invalid command
+     */
+    private static void printInvalidCommand() {
+        printBorder();
+        System.out.println("Command not recognized!");
+        System.out.println("try the following: \"list\", \"done\", \"todo\", \"deadline\", \"event\", \"bye\"");
+        System.out.println("\"delete\"");
+        printBorder();
+    }
+
+    /*
+     * ===============================================
+     *            Miscellaneous Functions
+     * ===============================================
+     */
+
+    /**
+     * Initialize Duke and generate taskList
+     */
+    private static void initDuke() {
+        taskList = new ArrayList<>();
+        File dataFile = new File("data");
+        dataFile.mkdir(); //create "data" folder if it does not exist
+        taskDataPath = "data/tasks.txt";
+        try {
+            loadFileContents(taskDataPath);
+            System.out.println("Previous data loaded successfully");
+        } catch (FileNotFoundException e) {
+            System.out.println("No existing data");
+        } catch (CorruptedDataException e) {
+            System.out.println("Data file corrupted...");
+        }
+    }
+
+    /**
+     * Tries to load an existing task data file
+     *
+     * @param filePath String representing the path of the data file
+     */
+    private static void loadFileContents(String filePath) throws FileNotFoundException, CorruptedDataException {
+        File f = new File(filePath);
+        Scanner dataScanner = new Scanner(f);
+        while (dataScanner.hasNext()) {
+            String[] data = dataScanner.nextLine().split(" \\| ");
+            switch (data[0]) {
+            case "T":
+                taskList.add(new Todo(data[2]));
+                if (data[1].equals("1")) {
+                    taskList.get(Task.getTotalTasks() - 1).setDone(true);
+                }
+                break;
+
+            case "D":
+                taskList.add(new Deadline(data[2], data[3]));
+                if (data[1].equals("1")) {
+                    taskList.get(Task.getTotalTasks() - 1).setDone(true);
+                }
+                break;
+
+            case "E":
+                taskList.add(new Event(data[2], data[3]));
+                if (data[1].equals("1")) {
+                    taskList.get(Task.getTotalTasks() - 1).setDone(true);
+                }
```

Comment:
Consider extracting the reused condition "data[1].equals("1")" into a boolean variable for better readability.